File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/messages/TransferInventoryAck.java
package com.lumiyaviewer.lumiya.slproto.messages;

import com.lumiyaviewer.lumiya.slproto.SLMessage;
import java.nio.ByteBuffer;
import java.util.UUID;

/**
 * dataserver -> sim
 * InventoryID is the id of the inventory object that the end user
 * should discard if they deny the transfer.
 *
 * <p>Template: {@code TransferInventoryAck Low 296 Trusted Zerocoded}
 * (recovered/reference/message_template.msg).
 */
public class TransferInventoryAck extends SLMessage {
    public InfoBlock InfoBlock_Field;

    /** Block InfoBlock, Single. */
    public static class InfoBlock {
        public UUID InventoryID; // LLUUID
        public UUID TransactionID; // LLUUID
    }

    public TransferInventoryAck() {
        this.zeroCoded = true;
        this.InfoBlock_Field = new InfoBlock();
    }

    @Override
    public int CalcPayloadSize() {
        return 36;
    }

    @Override
    public void Handle(SLMessageHandler messageHandler) {
        messageHandler.HandleTransferInventoryAck(this);
    }

    @Override
    public void PackPayload(ByteBuffer byteBuffer) {
        // Message number: Low 296 (TransferInventoryAck).
        byteBuffer.putShort((short) 0xFFFF);
        byteBuffer.put((byte) 0x01);
        byteBuffer.put((byte) 0x28);
        packUUID(byteBuffer, this.InfoBlock_Field.TransactionID);
        packUUID(byteBuffer, this.InfoBlock_Field.InventoryID);
    }

    @Override
    public void UnpackPayload(ByteBuffer byteBuffer) {
        this.InfoBlock_Field.TransactionID = unpackUUID(byteBuffer);
        this.InfoBlock_Field.InventoryID = unpackUUID(byteBuffer);
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/messages/TransferInventoryAck.kt
package com.lumiyaviewer.lumiya.slproto.messages

import com.lumiyaviewer.lumiya.slproto.SLMessage
import java.nio.ByteBuffer
import java.util.UUID

/**
 * dataserver -> sim
 * InventoryID is the id of the inventory object that the end user
 * should discard if they deny the transfer.
 *
 * <p>Template: {@code TransferInventoryAck Low 296 Trusted Zerocoded}
 * (recovered/reference/message_template.msg).
 */
open class TransferInventoryAck : SLMessage() {
    @JvmField var InfoBlock_Field: InfoBlock = InfoBlock()

    /** Block InfoBlock, Single. */
    open class InfoBlock {
        @JvmField var InventoryID: UUID? = null
        @JvmField var TransactionID: UUID? = null
    }

    init {
        zeroCoded = true
    }

    override fun CalcPayloadSize(): Int {
        return 36
    }

    override fun Handle(messageHandler: SLMessageHandler) {
        messageHandler.HandleTransferInventoryAck(this)
    }

    override fun PackPayload(byteBuffer: ByteBuffer) {
        // Message number: Low 296 (TransferInventoryAck).
        byteBuffer.putShort(0xFFFF.toShort())
        byteBuffer.put((0x01).toByte())
        byteBuffer.put((0x28).toByte())
        packUUID(byteBuffer, InfoBlock_Field.TransactionID)
        packUUID(byteBuffer, InfoBlock_Field.InventoryID)
    }

    override fun UnpackPayload(byteBuffer: ByteBuffer) {
        InfoBlock_Field.TransactionID = unpackUUID(byteBuffer)
        InfoBlock_Field.InventoryID = unpackUUID(byteBuffer)
    }
}
